libsolidity/analysis/TypeChecker.cpp

```diff
@@ -3632,6 +3632,27 @@ void TypeChecker::endVisit(Literal const& _literal)
 
 void TypeChecker::endVisit(UsingForDirective const& _usingFor)
 {
+	if (_usingFor.global())
+	{
+		if (m_currentContract)
+			solAssert(m_errorReporter.hasErrors());
+		if (Declaration const* typeDefinition = _usingFor.typeName()->annotation().type->typeDefinition())
+		{
+			if (typeDefinition->scope() != m_currentSourceUnit)
+				m_errorReporter.typeError(
+					4117_error,
+					_usingFor.location(),
+					"Can only use \"global\" with types defined in the same source unit at file level."
```

Comment:
Is something like this supposed to work?
`b.sol`
```solidity
import {S} from "./a.sol";

using L for S global;
library L {}
```
`a.sol`
```solidity
struct S { uint x; }
```

It currently compiles without errors on 0.8.13. If it is then this error message is not entirely correct.

Reply:
It should not work - here is a test: https://github.com/ethereum/solidity/blob/develop/test/libsolidity/syntaxTests/using/global_for_type_from_other_file.sol

Reply:
My example does compile on 0.8.13 though while `global_for_type_from_other_file.sol` does not. Apparently libraries are treated differently. Can you add my example and make sure it's reported as error?

Reply:
Or just add a library to that existing test.

Reply:
Added as "global_library_foreign.sol"